src/components/MindmapWrapper.vue

```diff
@@ -105,26 +105,31 @@ export default {
   watch: {
     items: {
       async handler() {
-        this.itemLists = this.items;
+        this.itemLists = structuredClone(this.items);
         let newMap = { ...this.emojiMenu };
         this.itemLists.map(async (item) => {
+          newMap[`menu-${item.stepID}`] = false;
           let temp = structuredClone(item);
-          newMap[`menu-${temp.stepID}`] = false;
-          if (this.getType(temp.fileType) === "mindmap")
-            temp.data = await this.$storageService.getItemById(temp.stepID);
+          if (this.getType(item.fileType) === "mindmap") {
+            if (this.$isElectron)
+              temp.data = await this.$storageService.getItemById(temp.stepID);
+            else {
+              const itemInStore = this.$store.state.session.items.find(
+                (newItem) => newItem.stepID === item.stepID
+              );
+              temp.data = structuredClone(itemInStore);
+            }
+          } else if (item.fileType === "audio/mp3") {
+            if (!this.$isElectron) {
+              temp.poster = this.generatePoster(item.filePath);
```

Comment:
can we generate these on the fly instead?

Reply:
if this is very slow, then maybe we shouldn't.  please look into it

Reply:
You know, generatePoster function generates the poster using audio file.
I think there is no need to generate posters everytime.
I think it's good to generate posters only once when creating the audio file.

Reply:
if we do it that way then we need to send the poster to and from the sever each time it is updated or accessed but if we generated it each time then we don't have to store the poster anywhere

Reply:
Yeah, so the previous code generates the poster each time when audio is updated and created.

Reply:
what about when it is displayed?  is it generated on the fly?

Reply:
Yes, when you update the audio, items will be changed.
And when items changed, it will call generatePoster function to generate the poster on the fly.
And the new poster will be displayed.

Reply:
when you pull an audio file from the backend that you have never had locally before and it tries to display on the timeline, does it generate on the fly?

Reply:
I think it doesn't now.
I will find the method to implement that and let you know.

Reply:
And who is currently responsible for the backend?

Reply:
Hello @dacoaster .
Currently on frontend side, it creates object url using blob (like this :  blob:http://localhost:8080/d6eba0d1-32a5 ) and it will be stored on the backend.
But you know, this blob url depends on the frontend. So, I mean if you refresh or close the web page, the blob url will be nothing.
So, I think you need to store the images and files to the backend in the other method.
If you do this, I can generate the posters on the fly with the audio files.

Reply:
@dockabernathy01 - the connection to the backend is brand new so many of the functions aren’t done yet.

currently the only “update” that exists is the [updateState](https://github.com/yatt-ai/yattie/blob/ok/integration/src/services/storage-options/restApiService.js#L12) method that sends the entire state to the backend.

it also handles [uploading files](https://github.com/yatt-ai/yattie/blob/ok/integration/src/services/storage-options/restApiService.js#L50)

if you are seeing errors there, please raise them as an issue on the backend repo and tag me there. Also include screenshots and error messages

I’m not sure what the issue you reference on the frontend is but it may go away if you fix your error on the backend

Reply:
@dacoaster , Ok, I will add issue on backend repo.
And I think you have seen my message above.
Currently frontend send blob urls as payload for updateState function.
So, I can't get the uploaded images and audios.
I think we need to update this method.

Reply:
This function is working for myself and others so let’s get your backend errors fixed first and see if that helps

Reply:
@dacoaster, Currently, when items changed, it generates poster in realtime.
Current generatePoster function generates poster using the audiofile (storage.googleapis.com/sandbox-attachments/.....).
So, there is no need to worry about the pulled audio file  from the backend.